source/MaterialXGenOgsXml/OgsFragment.cpp

```diff
@@ -0,0 +1,353 @@
+    {
+        // Use FIS environment lookup for surface shader generation but
+        // disable for texture nodes to avoid additional unneeded XML parameter
+        // generation.    
+        genOptions.hwSpecularEnvironmentMethod =
+            _isSurface ? mx::SPECULAR_ENVIRONMENT_FIS : mx::SPECULAR_ENVIRONMENT_NONE;
+
+        // Set to use no direct lighting
+        genOptions.hwMaxActiveLightSources = 0;
+
+        // Maya images require a texture coordinates to be flipped in V.
+        genOptions.fileTextureVerticalFlip = true;
+        genOptions.hwTransparency = mx::isTransparentSurface(_element, generator);
+
+        // Maya viewport uses texture atlas for tile image so enabled
+        // texture coordinate transform to go from original UDIM range to
+        // normalized 0..1 range.
+        genOptions.hwNormalizeUdimTexCoords = true;
+    }
+
+    mx::ElementPtr _element;
+
+private:
+    bool _isSurface = false;
+};
+
+// Knows how to create a temporary local GLSL fragment generator to generate
+// GLSL fragment code during OgsFragment construction.
+//
+class LocalGlslGeneratorWrapper
+    : public GlslGeneratorWrapperBase
+{
+public:
+    LocalGlslGeneratorWrapper(
+        mx::ElementPtr element,
+        const mx::FileSearchPath& librarySearchPath
+    )
+        : GlslGeneratorWrapperBase(element)
+        , _librarySearchPath(librarySearchPath)
+    {
+    }
+
+    ~LocalGlslGeneratorWrapper()
+    {}
+
+    std::pair<mx::ShaderPtr, bool> operator()(const std::string& baseFragmentName)
+    {
+        mx::ShaderGeneratorPtr generator = mx::GlslFragmentGenerator::create();
+        mx::GenContext genContext(generator);
+        mx::GenOptions& genOptions = genContext.getOptions();
+
+        // Set up color management. We assume the target render space is linear
+        // if not found in the document. Currently the default system has no other color space targets.
+        //
+        if ( mx::DefaultColorManagementSystemPtr colorManagementSystem =
+            mx::DefaultColorManagementSystem::create(generator->getLanguage())
+        )
+        {
+            generator->setColorManagementSystem(colorManagementSystem);
+
+            mx::DocumentPtr document = _element->getDocument();
+            colorManagementSystem->loadLibrary(document);
+            const std::string& documentColorSpace = document->getAttribute(mx::Element::COLOR_SPACE_ATTRIBUTE);
+
+            static const std::string MATERIALX_LINEAR_WORKING_SPACE("lin_rec709");
+            genOptions.targetColorSpaceOverride =
+                documentColorSpace.empty() ? MATERIALX_LINEAR_WORKING_SPACE : documentColorSpace;
+        }
+
+        genContext.registerSourceCodeSearchPath(_librarySearchPath);
+        
+        setCommonOptions(genOptions, *generator);
+
+        return std::make_pair(
+            generator->generate(baseFragmentName, _element, genContext),
+            genOptions.hwTransparency
```

Comment:
You might not need to return the transparency flag explicitly here. There's an attribute set on the shader if it includes fragments for transparency. So you can query: 
`shader->hasAttribute(HW::ATTR_TRANSPARENT)`

Reply:
Thanks for the tip, I was wondering if a mechanism like this existed.